src/Run/Setup.cs

```diff
@@ -150,10 +150,7 @@ private string BuildParametersForCommand(Dictionary<string, string> commandParam
         {
             string commandSetting = string.Empty;
 
-            if (Tools.ContainsKey(toolName))
-            {
-                commandSetting = Tools[toolName].osSpecific[Os]["defaultParameters"];
-            }
+            Tools[toolName].osSpecific[Os].TryGetValue("defaultParameters", out commandSetting);            
```

Comment:
Is it a valid assumption that Tools will always have the toolName entry? Below it seems like you are still guarding against that.